pkg/kubelet/server/stats/kubelet_stats_provider.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package stats
+
+import (
+	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+// KubeletStatsProvider provides access to statistics about resources managed by the kubelet
+type KubeletStatsProvider interface {
+	// ListVolumesForPod returns a map of volume-name to volume.Volume for all volumes defined in the Pod Spec
+	ListVolumesForPod(podUID types.UID) (map[string]volume.Volume, error)
```

Comment:
Why is this method not part of Kubelet core interface? It feels weird to define a new interface for listing volumes.